src/main/java/github/scarsz/discordsrv/listeners/PlayerAdvancementDoneListener.java

```diff
@@ -65,46 +66,82 @@ public void onPlayerAdvancementDone(PlayerAdvancementDoneEvent event) {
-                .replace("%username%", player.getName())
-                .replace("%displayname%", DiscordUtil.strip(DiscordUtil.escapeMarkdown(player.getDisplayName())))
-                .replace("%world%", player.getWorld().getName())
-                .replace("%achievement%", advancementName);
-        discordMessage = PlaceholderUtil.replacePlaceholdersToDiscord(discordMessage, event.getPlayer());
+        messageFormat = preEvent.getMessageFormat();
+
+        if (messageFormat == null) return;
+
+        String finalAchievementName = StringUtils.isNotBlank(advancementName) ? advancementName : "";
+        String avatarUrl = DiscordSRV.getPlugin().getEmbedAvatarUrl(player);
+        String botAvatarUrl = DiscordUtil.getJda().getSelfUser().getEffectiveAvatarUrl();
+        String botName = DiscordSRV.getPlugin().getMainGuild() != null ? DiscordSRV.getPlugin().getMainGuild().getSelfMember().getEffectiveName() : DiscordUtil.getJda().getSelfUser().getName();
+        String webhookName = messageFormat.getWebhookName();
+        String webhookAvatarUrl = messageFormat.getWebhookAvatarUrl();
+        String displayName = StringUtils.isNotBlank(player.getDisplayName()) ? player.getDisplayName() : "";
+
+        String finalChannelName = channelName;
+        BiFunction<String, Boolean, String> translator = (content, needsEscape) -> {
+            if (content == null) return null;
+            content = content
+                    .replaceAll("%time%|%date%", TimeUtil.timeStamp())
+                    .replace("%username%", player.getName())
+                    .replace("%displayname%", DiscordUtil.strip(needsEscape ? DiscordUtil.escapeMarkdown(displayName) : displayName))
+                    .replace("%world%", player.getWorld().getName())
+                    .replace("%achievement%", DiscordUtil.strip(needsEscape ? DiscordUtil.escapeMarkdown(finalAchievementName) : finalAchievementName))
+                    .replace("%embedavatarurl%", avatarUrl)
+                    .replace("%botavatarurl%", botAvatarUrl)
+                    .replace("%botname%", botName);
+            TextChannel textChannel = DiscordSRV.getPlugin().getDestinationTextChannelForGameChannelName(finalChannelName);
+            if (textChannel != null) content = DiscordUtil.translateEmotes(content, textChannel.getGuild());
+            content = PlaceholderUtil.replacePlaceholdersToDiscord(content, player);
+            return content;
+        };
+        Message discordMessage = DiscordSRV.getPlugin().translateMessage(messageFormat, translator);
+        if (discordMessage == null) return;
+
+        webhookName = translator.apply(webhookName, true);
+        webhookAvatarUrl = translator.apply(webhookAvatarUrl, true);
+
+        if (advancementName.length() < 3) {
+            DiscordSRV.debug("Not sending achievement message, because it's less than three characters long. Message: " + messageFormat);
+            return;
+        }
```

Comment:
This check isn't necessary for achievements/advancements

Reply:
Removed